src/main/java/com/parasoft/findings/sonar/sensor/SOAtestTestExecutionSensor.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2024 Parasoft Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.parasoft.findings.sonar.sensor;
+
+import com.parasoft.findings.sonar.Logger;
+import com.parasoft.findings.sonar.Messages;
+import com.parasoft.findings.sonar.ParasoftConstants;
+import com.parasoft.findings.sonar.soatest.SurefireSOAtestParser;
+import com.parasoft.findings.sonar.soatest.TestExecutionReportConverter;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+
+import java.io.File;
+import java.util.List;
+
+public class SOAtestTestExecutionSensor implements Sensor {
+
+    private final TestExecutionReportConverter testExecutionReportConverter;
+
+    private final SurefireSOAtestParser surefireSOAtestParser;
+
+    public SOAtestTestExecutionSensor(TestExecutionReportConverter testExecutionReportConverter, SurefireSOAtestParser surefireSOAtestParser, FileSystem fs) {
+        this.testExecutionReportConverter = testExecutionReportConverter;
+        this.surefireSOAtestParser = surefireSOAtestParser;
+    }
+
+    @Override
+    public void describe(SensorDescriptor descriptor) {
+        descriptor.name(ParasoftConstants.PARASOFT_SOATEST_TEST_EXECUTION_IMPORTER);
```

Comment:
Will `onlyOnFileType` have the control over what files will be scanned and listed in the source editor view? If so, is there any way to list only tst files under Measure tab?

Reply:
We may try adding a language for tst files. [Narrowing the focus with an analysis scope](https://docs.sonarsource.com/sonarqube/latest/project-administration/analysis-scope/#scope-by-file-type)

Reply:
We can use the `-Dsonar.inclusions=**/*.tst` option to include only tst files. See: [Project level - Source File Inclusions - Narrowing the focus with an analysis scope](https://docs.sonarsource.com/sonarqube/10.3/project-administration/analysis-scope/#project-level).